tests/Unit/Cache/Psr16Test.php

```diff
@@ -0,0 +1,240 @@
+use PHPUnit\Framework\TestCase;
+use Psr\SimpleCache\CacheInterface;
+use SimplePie\Cache\Psr16;
+use SimplePie\Tests\Fixtures\Exception\Psr16CacheException;
+
+class Psr16Test extends TestCase
+{
+    public function testImplementationIsGloballyStored()
+    {
+        $this->expectException(Exception::class);
+        $this->expectExceptionMessage('You must set an implementation of `Psr\SimpleCache\CacheInterface` via `SimplePie\SimplePie::set_cache()` first.');
+
+        new Psr16('location', 'name', 'type');
+    }
+
+    public function testSaveReturnsTrue()
+    {
+        $data = [];
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->exactly(2))->method('set')->withConsecutive(
+            [
+                '18bb0a0a94b49eda35e8944672d60a1474ff2895524f0e56c3752c1e0f7853e7',
+                $data,
+                3600,
+            ],
+            [
+                '18bb0a0a94b49eda35e8944672d60a1474ff2895524f0e56c3752c1e0f7853e7_mtime'
+            ]
+        )->willReturn(true);
+
+        Psr16::store_cache($psr16);
+        $cache = new Psr16('location', 'name', 'type');
+
+        $this->assertTrue($cache->save($data));
+    }
+
+    public function testSaveReturnsFalse()
+    {
+        $data = [];
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->exactly(2))->method('set')->willReturn(false);
+
+        Psr16::store_cache($psr16);
+        $cache = new Psr16('location', 'name', 'type');
+
+        $this->assertFalse($cache->save($data));
+    }
+
+    public function testSaveCatchExceptionAndReturnsFalse()
+    {
+        $e = new Psr16CacheException();
+
+        $data = [];
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->once())->method('set')->willThrowException($e);
+
+        Psr16::store_cache($psr16);
+        $cache = new Psr16('location', 'name', 'type');
+
+        $this->assertFalse($cache->save($data));
+    }
+
+    public function testLoadReturnsData()
+    {
+        $data = [];
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->once())->method('get')->willReturn($data);
+
+        Psr16::store_cache($psr16);
+        $cache = new Psr16('location', 'name', 'type');
+
+        $this->assertSame($data, $cache->load());
+    }
+
+    public function testLoadReturnsFalse()
+    {
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->once())->method('get')->willReturnCallback(function ($a, $b) {
+            return $b;
+        });
+
+        Psr16::store_cache($psr16);
+        $cache = new Psr16('location', 'name', 'type');
+
+        $this->assertSame(false, $cache->load());
+    }
+
+    public function testMtimeReturnsCorrectInt()
+    {
+        $data = 1234568790;
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->once())->method('get')->willReturn($data);
```

Comment:
Maybe add a comment like “Modification time is stored as a separate cache key” and rename the `$data` variable to `$mtime`?

Reply:
Done.

Btw I used the same principle as in the Memcached class, see https://github.com/simplepie/simplepie/blob/1.7.0/src/Cache/Memcached.php#L167-L172